### PR TITLE
[JENKINS-73969] [JENKINS-73970] Remove inline JavaScript to become CSP compliant

### PR DESCRIPTION
**Pull request by mawinter69**
use checkDependsOn for checkUrl
move the script for UserRelevanceFilter to a separate js file and load as adjunct
fix that nameOptions were not hidden when when selecting a non name matcher in regex filter

### Testing done
Manual Testing

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jglick — APPROVED** [review]

File: src/main/resources/hudson/views/RegExJobFilter/config.jelly [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form"
 	xmlns:i="jelly:fmt" xmlns:p="/lib/hudson/project">
   <f:entry title="${%Regular Expression}:">
-      <f:textbox name="regex" field="regex" checkUrl="'${rootURL}/descriptorByName/hudson.views.RegExJobFilter/checkRegex?value='+encodeURIComponent(this.value)"/>
+    <f:textbox name="regex" field="regex" checkUrl="${rootURL}/descriptorByName/hudson.views.RegExJobFilter/checkRegex" checkDependsOn=""/>
```
Comment (basil) [inline src/main/resources/hudson/views/RegExJobFilter/config.jelly:6]: Thanks for your contribution to this plugin @mawinter69! Unfortunately, there is still one more CSP violation in this plugin after these changes: the `onchange` attribute a few lines down on line 11, which is an inline event handler. Would it be too much to ask for you to fix that up in this PR as well? If that is asking too much, feel free to say so, and in that case I will create a separate Jira ticket to cover that remaining violation.

Comment (mawinter69) [reply src/main/resources/hudson/views/RegExJobFilter/config.jelly:6]: I can look into this in this PR, no problem

Comment (mawinter69) [reply src/main/resources/hudson/views/RegExJobFilter/config.jelly]: done. Also fixed the functionality which was broken since 2 years

**Review by basil — APPROVED** [review]
Thank you very much!

**Pull request merged**